### ISSUE TITLE
Unnamed snapshots don't compile anymore

### ISSUE THREAD

**Issue by jonas-schievink** [mitsuhiko/insta#36]
```rust
assert_snapshot_matches!("string");
```
results in:
```
error[E0308]: mismatched types
   --> ta-server/src/bits.rs:209:9
    |
209 |         assert_snapshot_matches!("string");
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `insta::runtime::ReferenceValue`, found enum `std::option::Option`
    |
    = note: expected type `insta::runtime::ReferenceValue<'_>`
               found type `std::option::Option<std::string::String>`
```

It looks like the example in the crate-level docs wasn't being tested (it's marked as `ignore`). Instead it could have been marked as `no_run` so it would have still been compiled.

**Comment by mitsuhiko** [comment]
That's not intentional. I will fix this.

**Comment by max-sixty** [comment]
@mitsuhiko I can have a look at this. 

What's the best way to solve it? 
- Add another variant `Unnamed` to the `ReferenceValue` Enum to [match on](https://github.com/mitsuhiko/insta/blob/master/src/runtime.rs#L445-L484)? 
- Make [this macro branch](https://github.com/mitsuhiko/insta/blob/master/src/macros.rs#L305-L307) pass a different value?
- ...

**Comment by mitsuhiko** [comment]
Closed by #38
